dot/parachain/availability-store/availability_store.go

```diff
@@ -420,8 +415,12 @@ func branchesFromChunks(chunks [][]byte) (branches, error) {
 }
 
 // Run runs the availability store subsystem
-func (av *AvailabilityStoreSubsystem) Run(ctx context.Context, OverseerToSubsystem chan any,
-	SubsystemToOverseer chan any) {
+func (av *AvailabilityStoreSubsystem) Run(
+	ctx context.Context, cancel context.CancelFunc,
```

Comment:
Its common to only passes the context to the `Run` method but not the cancelation function, like:

```go
ctx, cancel := context.WithCancel(context.Background())
av.Run(ctx)
```

Does not make sense to give to the av the possibility to cancel the context it have, also since this context is used by `av.processMessages` I would suggest to pass the context to it then you don't need to keep it as a field in `AvailabilityStoreSubsystem` , for example:

```go
func (av *AvailabilityStoreSubsystem) Run(ctx context.Context) {
    ...
    go av.processMessages(ctx)
}

func (av *AvailabilityStoreSubsystem) processMessages(ctx context.Context) {
    ...
    for {
       select {
       case <-ctx.Done():
             ...
       }
    }
}
```